app/src/main/AndroidManifest.xml

```diff
@@ -11,7 +11,8 @@
         <activity
             android:name=".PuzzleSelection"
             android:label="@string/app_name"
-            android:theme="@style/AppTheme.ActionBar">
+            android:theme="@style/AppTheme.ActionBar"
+            android:exported="true">
```

Comment:
See https://developer.android.com/guide/topics/manifest/activity-element#exported

related to bump to `targetSdkVersion 31`